### PR TITLE
Base caching on TypeLoader

### PR DESCRIPTION
**Pull request by adamhathcock**

### CODE REVIEW

File: src/Speckle.Sdk/Models/DynamicBase.cs [hunk 1]
```diff
@@ -153,78 +151,54 @@ public bool IsPropNameValid(string name, out string reason)
...
   /// </summary>
   /// <returns></returns>
-  public IReadOnlyList<PropertyInfo> GetInstanceMembers()
+  public IEnumerable<PropertyInfo> GetInstanceMembers()
   {
     return GetInstanceMembers(GetType());
   }
 
-  public static IReadOnlyList<PropertyInfo> GetInstanceMembers(Type t)
+  public static IEnumerable<PropertyInfo> GetInstanceMembers(Type t)
```
Comment (JR-Morgan) [inline src/Speckle.Sdk/Models/DynamicBase.cs:192]: Unsure about this change, 
returning a fully enumerated list means consumers know the size (so they can pre-size their collections appropriately). we ourselves use this in [here](https://github.com/specklesystems/speckle-sharp-sdk/blob/dc4da4907851ef581083926da83f24be7b37f1eb/src/Speckle.Sdk/Serialisation/BaseObjectSerializerV2.cs#L500-L501)

Imo, I'd actually prefer we return correctly pre-sized `IReadOnlyList` everywhere in this class... (maybe except for the overrides)

Comment (adamhathcock) [reply src/Speckle.Sdk/Models/DynamicBase.cs:192]: I'd rather not do allocations unless needed.   Only one place needed it.

File: src/Speckle.Sdk/Serialisation/BaseObjectSerializerV2.cs [hunk 2]
```diff
@@ -491,13 +491,13 @@ private void StoreBlob(Blob obj)
...
   {
     Type type = baseObj.GetType();
-    IReadOnlyList<PropertyInfo> typedProperties = baseObj.GetInstanceMembers();
 
     if (_typedPropertiesCache.TryGetValue(type.FullName, out List<(PropertyInfo, PropertyAttributeInfo)>? cached))
     {
       return cached;
     }
 
+    var typedProperties = baseObj.GetInstanceMembers().ToList();
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply src/Speckle.Sdk/Serialisation/BaseObjectSerializerV2.cs:500]: Yes, it's slower but not all callers need to buffer like here

[the remark this answers is not in the document]
Comment (JR-Morgan) [reply src/Speckle.Sdk/Serialisation/BaseObjectSerializerV2.cs:500]: lets measure if this saves or costs allocations

**Review by JR-Morgan — APPROVED** [review]

**Pull request merged**